### PR TITLE
fixes index.json encoding problems with non-english characters

### PR DESCRIPTION
**Pull request by reinaldoarrosi**
It appears that the problem is caused when reading .html files used to generate the index.json file in ExtractSearchIndex class.

I've changed the encoding used to read the .html files to UTF-8 since, it appears to me, that .html files are always generated in UTF-8.

#650

### CODE REVIEW

File: test/Microsoft.DocAsCode.Build.Engine.Tests/ExtractSearchIndexFromHtmlTest.cs [hunk 1]
```diff
@@ -7,6 +7,10 @@ namespace Microsoft.DocAsCode.Build.Engine.Tests
 
     using HtmlAgilityPack;
     using Xunit;
+    using System.IO;
+    using Plugins;
+    using System.Collections.Generic;
```
Comment (vwxyzh) [inline test/Microsoft.DocAsCode.Build.Engine.Tests/ExtractSearchIndexFromHtmlTest.cs]: - Sort namespaces
- Use full name, e.g.: `Microsoft.DocAsCode.Plugins`

**Comment by vwxyzh** [comment]
:shipit:

**Comment by superyyrrzz** [comment]
:shipit:

**Comment by hellosnow** [comment]
:shipit:

**Comment by qinezh** [comment]
:shipit:

**Pull request merged**